### DISCUSSION TITLE
Cache is not available in cross script execution

### DISCUSSION THREAD

**Discussion by rahulradixweb** [TruCopilot/phpfastcache#836] [Q&A] [answered]
Hello,

We have one common file that is included in every execution say common.php with following code

common.php
```
$InstanceCache = Phpfastcache\CacheManager::getInstance('files');
    Phpfastcache\CacheManager::setDefaultConfig(new Phpfastcache\Config\ConfigurationOption([
        "path" => "/tmpcache/",
        "defaultTtl" => 2629746        
    ]));
```

from one file I set cache say, **file1.php that includes common.php like**

```
$cachedItem = $InstanceCache->get('test_key');
$cachedItem->set('test_data');
$InstanceCache->save($cachedItem);
```

Now I want to access that cached data in **file2.php that includes common.php like.**

`$cachedItem = $InstanceCache->get('test_key');`

But I get empty item as $InstanceCache is a new fresh instance,
Any views on this?
**### We want cached data to be accessed on cross script execution,** as for example we have same product listing fetched from database in separate pages, so If I visit first page, it should be fetched and cached, again I visit second page, data should be fetched from cache.

Your support is appriciated.
Thanks.

**Comment by Geolim4** [comment, marked as answer]
It's because you call `CacheManager::setDefaultConfig` after building your cache instance.
using `CacheManager::setDefaultConfig` is useful if you have multiple cache pool instances.

In your case you can do the following:

```
$InstanceCache = Phpfastcache\CacheManager::getInstance('files', new \Phpfastcache\Config\ConfigurationOption([
        "path" => "/tmpcache/",
        "defaultTtl" => 2629746        
    ]));
```

**Reply by rahulradixweb** [reply]
Thanks, It helps!

**Comment by Geolim4** [comment]
And please read carefully the [WIKI](https://github.com/PHPSocialNetwork/phpfastcache/wiki) and the [DOCS](https://github.com/PHPSocialNetwork/phpfastcache/tree/master/docs). You can also look at the way I wrote the [tests](https://github.com/PHPSocialNetwork/phpfastcache/tree/master/tests) too. There's many documentation you can use to implement Phpfastcache.
